src/components/Calendar/hooks/useDaysBuilder.js

```diff
@@ -0,0 +1,55 @@
+import React, { useMemo } from 'react';
+import { addDays, shouldDateBeSelected, isDateWithinRange, isSameDay } from '../helpers';
+
+export default function useDaysBuilder(Component, props) {
```

Comment:
you don't need to pass Component, it's better to import Day here, just pass all the props to it. We are also trying to make everything functional. You can change something like this:

`
function useDaysBuilder(
    startDate,
    endDate,
    firstDayMonth,
    lastDayMonth,
    minDate,
    maxDate,
    selectionType,
    value,
    selectedRange,
    currentRange,
    onChange,
    ) {
    return useMemo(() => {

        const daysDiff = endDate.getDate() - startDate.getDate();
        return Array.from(Array(daysDiff), (_, index) => {
            const date = addDays(startDate, index);
            return (
                <Day
                    date={date}
                    firstDayMonth={firstDayMonth}
                    key={date.getTime()}
                    minDate={minDate}
                    maxDate={maxDate}
                    onChange={onChange}
                    isSelected={shouldDateBeSelected(date, value, selectionType, selectedRange)}
                    isWithinRange={isDateWithinRange(date, currentRange)}
                    isFirstDayOfWeek={index === 0 || isSameDay(date, firstDayMonth)}
                    isLastDayOfWeek={index === 6 || isSameDay(date, lastDayMonth)}
                />
            );
        });
    }, [
        currentRange,
        endDate,
        firstDayMonth,
        lastDayMonth,
        maxDate,
        minDate,
        onChange,
        selectedRange,
        selectionType,
        startDate,
        value,
    ]);
}
`